components/about/RoadmapList.vue

```diff
@@ -0,0 +1,327 @@
+<template>
+  <section>
+    <v-container>
+      <v-stack direction="column">
+        <app-section-descriptor title="Roadmap" heading="Eyes on the Road" />
+        <v-stack gap="1rem" class="laptop-remove">
+          <v-dropdown
+            v-model="selectedYear"
+            :options="years"
+            class="roadmap-dropdown"
+          />
+          <v-dropdown
+            v-model="selectedQuarter"
+            :options="quarters"
+            class="roadmap-dropdown"
+          />
+        </v-stack>
+        <v-stack>
+          <v-stack
+            direction="column"
+            style="margin: 1.5rem 6rem 0 0"
+            class="tablet-remove mobile-remove"
+          >
+            <div v-for="year in years" :key="year">
+              <v-chip
+                class="clickable-chip"
+                @click="() => (selectedYear = year)"
+              >
+                <v-text :weight="600">{{ year }}</v-text>
+              </v-chip>
+              <ul v-if="selectedYear === year">
+                <li
+                  v-for="quarter in quarters"
+                  :key="quarter + year"
+                  class="quarter"
+                  :class="{
+                    'selected-quarter': selectedQuarter === quarter,
+                  }"
+                  @click.stop="() => (selectedQuarter = quarter)"
+                >
+                  <v-text>{{ quarter }}</v-text>
+                </li>
+              </ul>
+            </div>
+          </v-stack>
+          <ul>
+            <li
+              v-for="item in roadmap[selectedYear][selectedQuarter]"
+              :key="item.milestone"
+            >
+              <v-heading
+                :level="4"
+                :weight="600"
+                font="serif"
+                class="milestone-heading"
+              >
+                {{ item.month }}&nbsp;&nbsp;&nbsp;&nbsp;{{ selectedYear }}
+              </v-heading>
+              <v-text
+                color="secondary"
+                :weight="400"
+                class="milestone-description"
+              >
+                {{ item.milestone }}
+              </v-text>
+            </li>
+          </ul>
+        </v-stack>
+      </v-stack>
+    </v-container>
+  </section>
+</template>
+
+<script>
+import VDropdown from '../lib/VDropdown.vue'
+export default {
+  name: 'RoadmapList',
+  components: { VDropdown },
+  data() {
+    return {
+      years: ['2019', '2020', '2021', '2022'],
+      quarters: ['Q1', 'Q2', 'Q3', 'Q4'],
+      selectedYear: '2019',
+      selectedQuarter: 'Q1',
```

Comment:
Assigned initial values, which is updated onMounted, it is calculated based on current year and month

Reply:
We can also move the calculations to the data property to avoid the re-render after mount.

```
data() {
 const selectedYear = new Date().getFullYear().toString()
 const currentMonth = new Date().getMonth() + 1
 const currentQuarterIndex = parseInt(currentMonth / 4)
 const selectedQuarter = this.quarters[currentQuarterIndex]

  return {
    ...,
    selectedYear,
    selectedQuarter,
  }
}
```

Reply:
Ohh I wasn't aware of that, I'll do that

Reply:
@punaifang updated this